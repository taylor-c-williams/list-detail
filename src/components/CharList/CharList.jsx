import { Link } from 'react-router-dom';

export default function CharList({ characters }) {
  return (
    <>
      <ul aria-label="characters">
        {characters.map((character) => (
          <li key={character.name}>
            <Link to={`/detail/${character.id}`}>
              <fieldset>
                <img src={character.photoUrl} alt={character.name} />
                <p>{character.name}</p>
              </fieldset>
            </Link>
          </li>
        ))}
      </ul>
    </>
  );
}
