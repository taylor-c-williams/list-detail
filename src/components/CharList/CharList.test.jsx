import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import CharList from './CharList';

const characters = [
  {
    _id: '5cf5679a915ecad153ab68c8',
    allies: ['Royal Earthbender Guards'],
    enemies: ['Chin'],
    photoUrl:
      'https://vignette.wikia.nocookie.net/avatar/images/5/51/46th_Earth_King.png/revision/latest?cb=20130627160441',
    name: '46th Earth King',
    affiliation: ' Earth Kingdom Earth Kingdom Royal Family',
  },
  {
    _id: '5cf5679a915ecad153ab68ca',
    allies: ['Gyatso'],
    enemies: ['Ozai'],
    photoUrl:
      'https://vignette.wikia.nocookie.net/avatar/images/2/21/Into_The_Inferno_Aang.png/revision/latest?cb=20131009060746',
    name: 'Aang (games)',
    affiliation: ' Air Nomads Team Avatar',
  },
];
describe('Characters', () => {
  it('renders all characters', async () => {
    const container = render(
      <MemoryRouter initialEntries={['/']}>
        <CharList characters={characters} />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
