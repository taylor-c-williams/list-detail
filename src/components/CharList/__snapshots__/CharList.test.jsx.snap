// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Characters renders all characters 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        aria-label="characters"
      >
        <li>
          <a
            href="/detail/undefined"
          >
            <fieldset>
              <img
                alt="46th Earth King"
                src="https://vignette.wikia.nocookie.net/avatar/images/5/51/46th_Earth_King.png/revision/latest?cb=20130627160441"
              />
              <p>
                46th Earth King
              </p>
            </fieldset>
          </a>
        </li>
        <li>
          <a
            href="/detail/undefined"
          >
            <fieldset>
              <img
                alt="Aang (games)"
                src="https://vignette.wikia.nocookie.net/avatar/images/2/21/Into_The_Inferno_Aang.png/revision/latest?cb=20131009060746"
              />
              <p>
                Aang (games)
              </p>
            </fieldset>
          </a>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      aria-label="characters"
    >
      <li>
        <a
          href="/detail/undefined"
        >
          <fieldset>
            <img
              alt="46th Earth King"
              src="https://vignette.wikia.nocookie.net/avatar/images/5/51/46th_Earth_King.png/revision/latest?cb=20130627160441"
            />
            <p>
              46th Earth King
            </p>
          </fieldset>
        </a>
      </li>
      <li>
        <a
          href="/detail/undefined"
        >
          <fieldset>
            <img
              alt="Aang (games)"
              src="https://vignette.wikia.nocookie.net/avatar/images/2/21/Into_The_Inferno_Aang.png/revision/latest?cb=20131009060746"
            />
            <p>
              Aang (games)
            </p>
          </fieldset>
        </a>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
